executor/src/witgen/machines/double_sorted_witness_machine.rs

```diff
@@ -130,6 +130,13 @@ impl<'a, T: FieldElement> Machine<'a, T> for DoubleSortedWitnesses<T> {
         let mut is_read = vec![];
 
         for ((a, s), o) in std::mem::take(&mut self.trace) {
+            if let Some(prev_address) = addr.last() {
+                assert!(a >= *prev_address, "Expected addresses to be sorted");
+                if (a - *prev_address).to_degree() >= self.degree {
+                    log::error!("Jump in memory accesses between {prev_address:x} and {a:x} is larger than or equal to the degree {}! This will violate the constraints.", self.degree);
```

Comment:
Not sure if this should instead be a panic? Alternatively, we'd have to change the signature of `Machine::take_witness_col_values()` to return a result.